core/lib/spree/permission_sets/restricted_stock_transfer_management.rb

```diff
@@ -27,7 +27,8 @@ def activate!
             source_location_id: location_ids,
             destination_location_id: destination_location_ids
 
-          can :transfer, Spree::StockLocation, id: location_ids
+          can :transfer_to, Spree::StockLocation, id: location_ids
```

Comment:
I agree the naming is a little confusing but looking at `destination_location_ids` it's just `location_ids + [nil]` and since we're not interested in the `nil` value I just used `location_ids`. Would a separate method, something like `transfer_to_location_ids` and `transfer_from_location_ids` be in order? I think it might be a bit much since they'll both end up calling `location_ids` but I do see how it could be helpful in terms of customization...